src/OmniSharp.Roslyn.CSharp/Services/Intellisense/CompletionItemExtensions.cs

```diff
@@ -18,55 +19,54 @@ internal static class CompletionItemExtensions
         private const string OverrideCompletionProvider = "Microsoft.CodeAnalysis.CSharp.Completion.Providers.OverrideCompletionProvider";
         private const string ParitalMethodCompletionProvider = "Microsoft.CodeAnalysis.CSharp.Completion.Providers.PartialMethodCompletionProvider";
-        private const string Provider = nameof(Provider);
+        private const string ProviderName = nameof(ProviderName);
         private const string SymbolCompletionItem = "Microsoft.CodeAnalysis.Completion.Providers.SymbolCompletionItem";
         private const string SymbolCompletionProvider = "Microsoft.CodeAnalysis.CSharp.Completion.Providers.SymbolCompletionProvider";
         private const string SymbolKind = nameof(SymbolKind);
         private const string SymbolName = nameof(SymbolName);
         private const string Symbols = nameof(Symbols);
 
         private static MethodInfo _getSymbolsAsync;
+        private static readonly PropertyInfo _getProviderName;
 
         static CompletionItemExtensions()
         {
             var symbolCompletionItemType = typeof(CompletionItem).GetTypeInfo().Assembly.GetType(SymbolCompletionItem);
             _getSymbolsAsync = symbolCompletionItemType.GetMethod(GetSymbolsAsync, BindingFlags.Public | BindingFlags.Static);
+
+            _getProviderName = typeof(CompletionItem).GetProperty("ProviderName", BindingFlags.NonPublic | BindingFlags.Instance);
+            
         }
 
         public static bool IsObjectCreationCompletionItem(this CompletionItem item)
         {
             var properties = item.Properties;
-            return properties.TryGetValue(Provider, out var provider) && provider == ObjectCreationCompletionProvider;
+            return properties.TryGetValue(ProviderName, out var provider) && provider == ObjectCreationCompletionProvider;
         }
 
-        public static async Task<IEnumerable<ISymbol>> GetCompletionSymbolsAsync(this CompletionItem completionItem, IEnumerable<ISymbol> recommendedSymbols, Document document)
+        public static IEnumerable<ISymbol> GetCompletionSymbols(this CompletionItem completionItem, IEnumerable<ISymbol> recommendedSymbols, Document document)
         {
             var properties = completionItem.Properties;
 
             // for SymbolCompletionProvider, use the logic of extracting information from recommended symbols
-            if (properties.TryGetValue(Provider, out var provider) && provider == SymbolCompletionProvider)
+            var providerName = (string)_getProviderName.GetValue(completionItem);
+            if (providerName == SymbolCompletionProvider)
             {
                 return recommendedSymbols.Where(x => x.Name == properties[SymbolName] && (int)x.Kind == int.Parse(properties[SymbolKind])).Distinct();
             }
 
             // if the completion provider encoded symbols into Properties, we can return them
-            if (properties.ContainsKey(Symbols))
+            if (properties.ContainsKey(SymbolName))
```

Comment:
I thought most (all?) completion items will be created via `CreateWithSymbolId`? Roslyn doesn't encode the symbol into the completion item anymore?

but if that's the case, we can make this return the same as above - `return recommendedSymbols.Where(x => x.Name == properties[SymbolName] && (int)x.Kind == int.Parse(properties[SymbolKind])).Distinct();`